app/controllers/admin_controller.rb

```diff
@@ -1,6 +1,7 @@
 class AdminController < ApplicationController
 
   def index
-    @top_customers = Customer.all_successful_transactions_count.order('successful_transactions_count desc').limit(5)
+    @top_customers = Customer.all_successful_transactions_by_customer_count.order('successful_transactions_count desc').limit(5)
```

Comment:
I asked our instructors about this, and they confirmed these lines are breaking MVC due to ActiveRecord in the controller.

Could you push the `order` and `limit` clauses into the Model, so the controller only has to call a model method?

Reach out on slack if you feel like there's anything you'd like to talk about -- I'm happy to help if I can.

Reply:
Thanks for the great discussion on this today. I have refactored both admin controller instance variables to have no AR methods

Reply:
Awesome!